frontend/test/resolution/testReturnTypes.cpp

```diff
@@ -1094,6 +1094,38 @@ static void testSelectParams() {
   }
 }
 
+static void testCPtrEltType() {
+  { 
+    //works for c_ptr
+    std::string program = ops + R"""(
+    var y: c_ptr(uint(8));
+    type x = y.eltType;
+    )""";
+
+    Context ctx;
+    Context* context = &ctx;
+    ErrorGuard guard(context);
+    auto qt = resolveTypeOfXInit(context, program);
+    assert(qt.type()->isUintType());
```

Comment:
Since you have `uint(8)` in the test file, consider validating the width as well in your assertions below.